tools/crm_mon.c

```diff
@@ -2050,18 +2033,7 @@ crm_diff_update(const char *event, xmlNode * msg)
     }
 
     if (options.external_agent) {
-        int format = 0;
-        crm_element_value_int(diff, "format", &format);
-        switch(format) {
-            case 1:
-                crm_diff_update_v1(event, msg);
-                break;
-            case 2:
-                crm_diff_update_v2(event, msg);
-                break;
-            default:
-                crm_err("Unknown patch format: %d", format);
-        }
+        handle_diff_for_external_agent(event, msg);
```

Comment:
It would still be useful to check the format and error if not 2. If there's ever a v3, this could be on an older remote node talking to a newer cluster node. Hopefully we will have some way of knowing not to send v3 diffs in that case, but it's always good to have a failsafe.